yield_plugin/src/yield_plugin_node.cpp

```diff
@@ -0,0 +1,124 @@
+    lc_status_pub_ = create_publisher<carma_planning_msgs::msg::LaneChangeStatus>("cooperative_lane_change_status", 10);
+
+    // Subscriber
+    mob_request_sub_ = create_subscription<carma_planning_msgs::msg::TrajectoryPlan>("incoming_mobility_request", 5,std::bind(&YieldPlugin::mobilityrequest_cb,&worker_,std_ph::_1));
+    bsm_sub_ = create_subscription<carma_planning_msgs::msg::TrajectoryPlan>("bsm_outbound", 1,std::bind(&YieldPlugin::bsm_cb,&worker_,std_ph::_1));
+    georeference_sub_ = create_subscription<carma_planning_msgs::msg::TrajectoryPlan>("georeference", 1,std::bind(&YieldPlugin::georeferenceCallback,&worker_,std_ph::_1));
+   
+    config_ = YieldPluginNode();
+
+    get_parameter<double>("acceleration_adjustment_factor", config_.acceleration_adjustment_factor);
+    get_parameter<double>("min_obstacle_speed", config_.min_obstacle_speed);
+    get_parameter<double>("collision_horizon", config_.collision_horizon);
+    get_parameter<double>("x_gap", config_.x_gap);
+    get_parameter<double>("max_stop_speed", config_.max_stop_speed);
+    get_parameter<double>("enable_cooperative_behavior", config_.enable_cooperative_behavior);
+    get_parameter<double>("always_accept_mobility_request", config_.always_accept_mobility_request);
+    get_parameter<double>("acceptable_passed_timesteps", config_.acceptable_passed_timesteps);
+    get_parameter<double>("intervehicle_collision_distance", config_.intervehicle_collision_distance);
+
+    get_parameter<double>("safety_collision_time_gap", config_.safety_collision_time_gap);
+    get_parameter<double>("enable_adjustable_gap", config_.enable_adjustable_gap);
+    get_parameter<double>("acceptable_urgency", config_.acceptable_urgency);
+    get_parameter<double>("speed_moving_average_window_size", config_.speed_moving_average_window_size);
+    get_parameter<double>("tpmin", config_.tpmin);
+    get_parameter<double>("/vehicle_length", config_.vehicle_length);
+    get_parameter<double>("/vehicle_height", config_.vehicle_height);
+    get_parameter<double>("/vehicle_width", config_.vehicle_width);
+    get_parameter<double>("/vehicle_id", config_.vehicle_id);
+
+    RCLCPP_INFO_STREAM(rclcpp::get_logger("yield_plugin"), "YieldPlugin Params" << config_);
+
+    carma_wm::WMListener wml;
+    auto wm_ = wml.getWorldModel();
+
+    worker_ = std::make_shared<YieldPlugin>(shared_from_this(), get_world_model(), config_,
+                                                          [&mob_resp_pub_](auto msg) { mob_resp_pub_->publish(msg); },
+                                                          [&lc_status_pub_](auto msg) { lc_status_pub_->publish(msg); },
+                                                          plugin_name_,
+                                                          version_id_);
+
+    trajectory_srv_ = create_client<carma_planning_msgs::srv::PlanTrajectory>("yield_plugin/plan_trajectory");
```

Comment:
I think this was not in the original yield_plugin's file, and just copy/paste error from other package right? If so, we don't need it

Reply:
because we need server on the plugins themselves, and the servers are being created by the base class.